src/CMakeLists.txt

```diff
@@ -64,20 +64,43 @@ foreach(kind ${kinds})
   set(lib_name ${PROJECT_NAME}_${kind})
   set(module_dir "${CMAKE_CURRENT_BINARY_DIR}/include_${kind}")
 
-  add_library(${lib_name} STATIC ${fortran_src} $<TARGET_OBJECTS:c_obj_lib>)
-  add_library(${PROJECT_NAME}::${lib_name} ALIAS ${lib_name})
-
-  set_target_properties(${lib_name} PROPERTIES COMPILE_FLAGS
+  # Create object library. The object library will be used to build
+  # static and shared libraries.
+#  add_library(${lib_name} STATIC ${fortran_src} $<TARGET_OBJECTS:c_obj_lib>)
+  add_library(${lib_name}_objlib OBJECT ${fortran_src} $<TARGET_OBJECTS:c_obj_lib>)
+  if(BUILD_SHARED_LIBS)
+    set_property(TARGET ${lib_name}_objlib PROPERTY POSITION_INDEPENDENT_CODE 1)
+  endif()
+  add_library(${PROJECT_NAME}::${lib_name} ALIAS ${lib_name}_objlib)
+
+  if(BUILD_STATIC_LIBS)
+    # Create static object library from object library.
+    add_library(${lib_name}_static STATIC $<TARGET_OBJECTS:${lib_name}_objlib>)
+    set_target_properties(${lib_name}_static PROPERTIES OUTPUT_NAME ${lib_name})
+    set_target_properties(${lib_name}_static PROPERTIES EXPORT_NAME ${lib_name})
+    set_target_properties(${lib_name}_static PROPERTIES EXPORT_NAME ${lib_name}_static)
+    list(APPEND LIB_TARGETS ${lib_name}_static)
+  endif()
+
+  if(BUILD_SHARED_LIBS)
+    # Create shared object library from object library.
+    add_library(${lib_name}_shared SHARED $<TARGET_OBJECTS:${lib_name}_objlib>)
+    set_target_properties(${lib_name}_shared PROPERTIES OUTPUT_NAME ${lib_name})
+    set_target_properties(${lib_name}_shared PROPERTIES EXPORT_NAME ${lib_name})
+    set_target_properties(${lib_name}_shared PROPERTIES SOVERSION 0)
```

Comment:
Add line after here...

`target_link_libraries(${lib_name}_shared PUBLIC bacio::${bacio_name})`

Reply:
Now when I compile I get this error:
/usr/bin/ld: /usr/local/NCEPLIBS-bacio/lib/libbacio.a(baciof.F90.o): relocation R_X86_64_PC32 against symbol `__bacio_module_MOD_fd' can not be used when making a shared object; recompile with -fPIC


Do I have to apply all these changes to bacio first?

Reply:
I think you are correct.  The dependent libraries would need to have the same capability for static and shared libs.  That is why I went after sp before ip.

I could take a shot at bacio if needed.